src/Traits/CommonMutableContainerTrait.php

```diff
@@ -12,56 +12,67 @@
     use ExtractTrait;
 
     /**
-     * {@inheritdoc}
+     * @return ArrayList
      */
     public function values()
     {
         return new ArrayList($this);
     }
 
+    /**
+     * @return array
+     */
     public function toValuesArray()
     {
         return $this->toArray();
     }
 
+    /**
+     * @return array
+     */
     public function toKeysArray()
     {
-        $res = [];
-        foreach ($this as $k => $_) {
-            $res[] = $k;
+        $results = [];
+        /** @var Dictionary $this */
```

Comment:
This is not going to be a dictionary all the time, we can have deferents implementations. Maybe using a *\Traversable* here?